src/app/dataValidator/helpers/dataLoader/readdirSync.test.js

```diff
@@ -6,14 +6,17 @@ const readScenario = require('./readScenario');
 
 const defaultDataPath = './data';
 
-const expectMethodToBeCalledTimes = (
+const expectMethodToBeCalledTimes = async (
   number,
   spyMethod,
   dataPath = defaultDataPath,
 ) => {
-  readdirSync(dataPath).then(() => {
+  await readdirSync(dataPath);
```

Comment:
`readdirSync` is synchronous - why would it need `await`ing?

Reply:
```
module.exports.readdirSync = dirName => {
  const fileNames = fs.readdirSync(dirName);

  return Promise.all(
    fileNames.map(fileName => readScenario.readScenario(fileName, dirName)),
  );
};
```

On review the method name is misleading, the function `readdirSync` actually does 2 things, synchronously reads the directory and then asynchronously validates each of the payloads returned from the synchronous directory read. 

Would renaming the method to `asyncValidateFilesInDirectory` clarify it's behaviour?

Reply:
Ha, interesting! `readdirSync` -> `readdirAsync` would probably be fine.

Any reason why we've written our own implementation and not used the [native Node methods](https://nodejs.org/api/fs.html#fs_fs_readdirsync_path_options) or a helper like [fs-extra](https://www.npmjs.com/package/fs-extra)? The less we have to write & test ourselves, the better IMO - I can raise an issue to refactor these out in the future...?

Reply:
Haven't looked into `fs-extra` before and IIRC we aren't using `fs.readdir` as it is easier to test the downstream `readScenario.readScenario` function by passing it an array rather than mocking a directory.